bin/update-package.sh

```diff
@@ -60,14 +60,18 @@ for package in packages/*; do
 
 	cp -r $BASE/packages/$NAME/. .
 
-	# Commit if there is anything to
-	if [ -n "$(git status --porcelain)" ]; then
+	# Before we commit any changes, ensure that the repo has the basics we need for any package.
+	if [ ! -f "composer.json" -o ! -d "src" ]; then
```

Comment:
you could also use `||` for "or" here, a bit more familiar to most people but not a blocker

Reply:
We can also use composer's `validate` command that checks for composer.json validity (and existence), as well as checking valid dependency versions and whether the `composer.lock` is up to date.

Sort of like this, for a `composer.lock` file not up to date:
```
$ composer validate -nA # -n for no interactive output -A for checking dependencies too
./composer.json is valid, but with a few warnings
See https://getcomposer.org/doc/04-schema.md for details on the schema
The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update`.
```

Or like this for a JSON error inside `composer.json`:
```
 composer validate -nA

In JsonFile.php line 304:
                                                 
  "./composer.json" does not contain valid JSON  
  Parse error on line 11:                        
  ...xt-fileinfo": "*"		"ext-json": "*",		"      
  ---------------------^                         
  Expected one of: 'EOF', '}', ':', ',', ']'     
                                                 
```

Reply:
@zinigor Do you think we'd need to add that check to the action? The pre-commit hook in the monorepo already prevents you from committing changes that break composer.json, so we should be okay there imo.

Reply:
Ah, I didn't realize we had that in the precommit hook. So this situation checks against a commit that skips pre-commit somehow?

Reply:
True, your idea would protect us against someone skipping the pre-commit hook.

I've started working on this in #14971.